service/matching/matcher_data.go

```diff
@@ -629,5 +718,19 @@ func (s *simpleLimiter) consume(now int64, tokens int64) {
 	//
 	// Alternatively, if now is > ready by more than burst, then we end up subtracting the full
 	// burst from now and adding one interval.
-	s.ready = max(now, s.ready+s.burst.Nanoseconds()) - s.burst.Nanoseconds() + tokens*s.interval.Nanoseconds()
+	if p.never() {
+		return simpleLimiterNever
+	}
+	clippedReady := max(now, int64(ready)+p.burst.Nanoseconds()) - p.burst.Nanoseconds()
+	return simpleLimiter(clippedReady + tokens*p.interval.Nanoseconds())
+}
+
+// clip sets ready to an allowable range based on the given parameters.
+func (ready simpleLimiter) clip(p simpleLimiterParams, now int64, maxTokens int64) simpleLimiter {
+	if p.interval < 0 {
+		return simpleLimiterNever
+	}
+	// if ready was set very far in the future (e.g. because the rate was zero), then we can
+	// clip it back to now + maxTokens*interval + burst.
+	return min(ready, simpleLimiter(now+maxTokens*p.interval.Nanoseconds()+p.burst.Nanoseconds()))
```

Comment:
Would it be helpful to extract a variable from `now+maxTokens*p.interval.Nanoseconds()+p.burst.Nanoseconds()` to give it a name?